### ISSUE TITLE
The negative_prompt is having little to no effect with the inpaint pipeline.

### ISSUE THREAD

**Issue by radi-cho** [huggingface/diffusers#1821]
### Describe the bug

It seems like `negative_prompt` is having little to no effect on the inpainting pipeline. Demonstrated with the example below.

Input:

![image](https://user-images.githubusercontent.com/12954909/209432599-f31194d2-904e-4179-96e7-c822ac9950fc.png)

Mask:

![image](https://user-images.githubusercontent.com/12954909/209432607-d43774d2-4712-4a24-8a14-a9ef55e01dce.png)

Parameters:

```
prompt="cloudy background"
seed=2345
num_outputs=1
guidance_scale=7.5
num_inference_steps=50

negative_prompt1=None
negative_prompt2="blue"
```

Result 1:

![image](https://user-images.githubusercontent.com/12954909/209432787-2f031d08-bce4-45db-8a80-018a0225faf8.png)

Result 2:

![image](https://user-images.githubusercontent.com/12954909/209432840-c50b0d3f-0859-4f0e-a7fe-a3565ae02596.png)

### Reproduction

```python
import torch
from PIL import Image
from diffusers import StableDiffusionInpaintPipeline

MODEL_ID = "stabilityai/stable-diffusion-2-inpainting"
MODEL_CACHE = "diffusers-cache"

pipe = StableDiffusionInpaintPipeline.from_pretrained(
    MODEL_ID,
    cache_dir=MODEL_CACHE,
    local_files_only=True,
).to("cuda")

image, mask, prompt, negative_prompt, seed, guidance_scale, num_inference_steps = ...

image = Image.open(image).convert("RGB").resize((512, 512))
mask = Image.open(mask).convert("RGB").resize(image.size)

generator = torch.Generator("cuda").manual_seed(seed)

output = pipe(
    prompt=[prompt] * num_outputs,
    negative_prompt=[negative_prompt] * num_outputs,
    guidance_scale=guidance_scale,
    generator=generator,
    num_inference_steps=num_inference_steps,
    mask_image=mask,
    image=image
)
```

### System Info

- `diffusers` version: 0.12.0.dev0
- Platform: Linux-4.19.0-22-cloud-amd64-x86_64-with-glibc2.28
- Python version: 3.9.0
- PyTorch version (GPU?): 1.13.1+cu116 (True)
- Huggingface_hub version: 0.11.1
- Transformers version: 4.25.1

**Comment by radi-cho** [comment]
Seems dependent on the number of inference steps. I was able to get the following image with 25 steps and the negative prompt "blue" (same seed and params as the original post).

![image](https://user-images.githubusercontent.com/12954909/209972382-51c4cdef-7290-4711-a57b-04210256d829.png)

**Comment by patrickvonplaten** [comment]
Interesting! I haven't tested inpainting + negative prompt too much yet. Maybe someone from the community has some idea here :-) 

I checked the code though and the code looks good, don't really think there is a bug in the code.

I know that negative prompt helps a lot with text2image, but should it also help a lot with image2image?
